src/UI/Internal/RenderConfig.elm

```diff
@@ -0,0 +1,39 @@
+module UI.Internal.RenderConfig exposing (Locale(..), RenderConfig(..), RenderConfigData, string, strings)
+
+import Element
+import I18n.English as LangEnglish
+import I18n.Spanish as LangSpanish
+import I18n.Type as I18n
+
+
+type alias RenderConfigData =
+    { deviceClass : Element.DeviceClass
+    , deviceOrientation : Element.Orientation
+    , locale : Locale
+    }
+
+
+{-| `RenderConfig.RenderConfig` upholds all the information required for the components to apply the responsivity and accessibility changes.
+-}
+type RenderConfig
+    = RenderConfig RenderConfigData
+
+
+type Locale
+    = EnglishGB
+    | SpanishES
+
+
+string : (I18n.Root -> String) -> RenderConfig -> String
+string getter cfg =
+    getter (strings cfg)
+
+
+strings : RenderConfig -> I18n.Root
```

Comment:
I think I prefer the word term/terms here 🤔  as string/strings are quite generic and may be confusing depending of the context we will use.

Reply:
or maybe localeStrings/localeTerms

Reply:
too much pollution when writing:
`(RenderConfig.localeStrings renderConfig).fields.period.before`